x/ibc/02-client/client/cli/tx.go

```diff
@@ -0,0 +1,155 @@
+	)...)
+
+	return ics02ClientTxCmd
+}
+
+// GetCmdCreateClient defines the command to create a new IBC Client as defined
+// in https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#create
+func GetCmdCreateClient(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [client-id] [path/to/consensus_state.json]",
+		Short: "create new client with a consensus state",
+		Long: strings.TrimSpace(fmt.Sprintf(`create new client with a specified identifier and consensus state:
+
+Example:
+$ %s tx ibc client create [client-id] [path/to/consensus_state.json] --from node0 --home ../node0/<app>cli --chain-id $CID
+		`, version.ClientName),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			contents, err := ioutil.ReadFile(args[1])
+			if err != nil {
+				return err
+			}
+
+			var state exported.ConsensusState
+			if err := cdc.UnmarshalJSON(contents, &state); err != nil {
+				return err
+			}
+
+			msg := types.MsgCreateClient{
+				ClientID: args[0],
+
+				ConsensusState: state,
+				Signer:         cliCtx.GetFromAddress(),
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
+
+// GetCmdUpdateClient defines the command to update a client as defined in
+// https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#update
+func GetCmdUpdateClient(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update [client-id] [path/to/header.json]",
+		Short: "update existing client with a header",
+		Long: strings.TrimSpace(fmt.Sprintf(`update existing client with a header:
+
+Example:
+$ %s tx ibc client create [client-id] [path/to/header.json] --from node0 --home ../node0/<app>cli --chain-id $CID
+		`, version.ClientName),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			bz, err := ioutil.ReadFile(args[1])
+			if err != nil {
+				return err
+			}
+
+			var header exported.Header
+			if err := cdc.UnmarshalJSON(bz, &header); err != nil {
+				return err
+			}
+
+			msg := types.MsgUpdateClient{
+				ClientID: args[0],
+				Header:   header,
+				Signer:   cliCtx.GetFromAddress(),
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
+
+// GetCmdSubmitMisbehaviour defines the command to submit a misbehaviour to invalidate
+// previous state roots and prevent future updates as defined in
+// https://github.com/cosmos/ics/tree/master/spec/ics-002-client-semantics#misbehaviour
+func GetCmdSubmitMisbehaviour(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "misbehaviour [client-it] [path/to/evidence.json]",
```

Comment:
This command should take either:
- The `json` emitted by `gaiacli q ibc client evidence`
- A file path to the data emitted by `gaiacli q ibc client evidence`

Maybe something like:

```go
			var evidence exported.Evidence
			if err := cdc.UnmarshalJSON([]byte(args[1]), &evidence); err != nil {
				fmt.Fprintf(os.Stderr, "failed to unmarshall input into struct, checking for file...")
				contents, err := ioutil.ReadFile(args[1])
				if err != nil {
					return fmt.Errorf("error opening evidence file: %v\n", err)
				}
				if err := cdc.UnmarshalJSON(contents, &evidence); err != nil {
					return fmt.Errorf("error unmarshalling evidence file: %v\n", err)
				}
			}
```